src/lib/functions/runtimes/js/builders/zisi.mjs

```diff
@@ -109,26 +125,44 @@ const getTargetDirectory = async ({ errorExit }) => {
 const netlifyConfigToZisiConfig = ({ config, projectRoot }) =>
   addFunctionsConfigDefaults(normalizeFunctionsConfig({ functionsConfig: config.functions, projectRoot }))
 
-export default async function handler({ config, directory, errorExit, func, projectRoot }) {
+export default async function handler({ config, directory, errorExit, func, metadata, projectRoot }) {
   const functionsConfig = netlifyConfigToZisiConfig({ config, projectRoot })
 
   const packageJson = await readPackageUp(func.mainFile)
   const hasTypeModule = packageJson && packageJson.packageJson.type === 'module'
 
-  // We must use esbuild for certain file extensions.
-  const mustTranspile = ['.mjs', '.ts'].includes(path.extname(func.mainFile))
-  const mustUseEsbuild = hasTypeModule || mustTranspile
-
-  if (mustUseEsbuild && !functionsConfig['*'].nodeBundler) {
-    functionsConfig['*'].nodeBundler = 'esbuild'
-  }
-
-  // TODO: Resolve functions config globs so that we can check for the bundler
-  // on a per-function basis.
-  const isUsingEsbuild = ['esbuild_zisi', 'esbuild'].includes(functionsConfig['*'].nodeBundler)
-
-  if (!isUsingEsbuild) {
-    return false
+  /** @type {import("@netlify/zip-it-and-ship-it/dist/feature_flags.js").FeatureFlags} */
+  const featureFlags = {}
+
+  if (metadata.runtimeAPIVersion === 2) {
+    // For TypeScript we use NFT, otherwise we leave the file untouched with the `none` bundler
+    const isTypescript = ['.ts', '.mts', '.cts'].includes(path.extname(func.mainFile))
+
+    if (isTypescript) {
+      functionsConfig['*'].nodeBundler = 'nft'
+    } else {
+      // using esbuild is less performant than `none`, but it emits sourcemaps and thus
```

Comment:
> We can only use none if we know upfront the list of files that need to be included, which isn't the case here?

Yes, because we want to copy over the source files. But if we didn't need to copy them in the first place (see https://github.com/netlify/cli/pull/5804#issuecomment-1607359866), then we wouldn't need to know the list.

Reply:
Ah, got you!